Updated some examples [MEP12]

adasilva opened the pull request:
Cleaned up examples: animate_decay.py, double_pendulum_animated.py, and barh_demo.py

tacaswell commented on examples/animation/animate_decay.py:
```diff
@@ -1,17 +1,29 @@
+    Generates a decaying sine wave.
+    """
+    count = 0
+    while count < 1000:
```
Can you change this to a `for count in range(1000):`  That is slightly more pythonic.

tacaswell commented on examples/animation/double_pendulum_animated.py:
```diff
@@ -86,5 +97,4 @@ def animate(i):
 ani = animation.FuncAnimation(fig, animate, np.arange(1, len(y)),
                               interval=25, blit=True, init_func=init)
 
-#ani.save('double_pendulum.mp4', fps=15)
```
Can you leave this line?  It is useful to show _how_ to save a movie even if we do not want to do it as part of building the docs.

adasilva replied:
Sure - I see how that could be useful. Keep as a comment?

WeatherGod replied:
yes

On Thu, May 26, 2016 at 9:55 PM, Ashley DaSilva <notifications@github.com>
wrote:

> In examples/animation/double_pendulum_animated.py
> <https://github.com/matplotlib/matplotlib/pull/6486#discussion_r64848024>:
>
> > @@ -86,5 +97,4 @@ def animate(i):
> >  ani = animation.FuncAnimation(fig, animate, np.arange(1, len(y)),
> >                                interval=25, blit=True, init_func=init)
> >
> > -#ani.save('double_pendulum.mp4', fps=15)
>
> Sure - I see how that could be useful. Keep as a comment?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/6486/files/0bd23dcdf6e696b67318eab42fabfff0be41e1b9#r64848024>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AARy-NhveMgFHXFuPt8O6sicISR_Y-7Nks5qFk8hgaJpZM4IoFO_>
> .
>

tacaswell commented on examples/lines_bars_and_markers/barh_demo.py:
```diff
@@ -8,7 +8,7 @@
 plt.rcdefaults()
 
 # Example data
-people = ('Tom', 'Dick', 'Harry', 'Slim', 'Jim')
```
:+1:

QuLogic commented on examples/animation/double_pendulum_animated.py:
```diff
@@ -1,38 +1,41 @@
-               M2*G*sin(state[2])*cos(del_) +
-               M2*L2*state[3]*state[3]*sin(del_) -
-               (M1 + M2)*G*sin(state[0]))/den1
+    den1 = (mass1 + mass2)*length1 - mass2*length1*np.cos(del_)*np.cos(del_)
```
While you are here, please add relevant spaces around operators.

QuLogic commented on examples/animation/double_pendulum_animated.py:
```diff
@@ -70,12 +73,20 @@ def derivs(state, t):
     line.set_data([], [])
     time_text.set_text('')
+    plt.xlabel('horizontal direction (m)')
+    plt.ylabel('vertical direction (m)')
```
I think these should say position instead of direction.

adasilva commented:
I made the requested changes.

afvincent commented:
For the PEP8-related failures, I think Travis is simply complaining about this rule “Don't use spaces around the = sign when used to indicate a keyword argument or a default parameter value.” (see [http://legacy.python.org/dev/peps/pep-0008/#other-recommendations](http://legacy.python.org/dev/peps/pep-0008/#other-recommendations))

tacaswell commented on examples/animation/double_pendulum_animated.py:
```diff
@@ -53,38 +59,46 @@ def derivs(state, t):
 
 ani = animation.FuncAnimation(fig, animate, np.arange(1, len(y)),
-                              interval=25, blit=True, init_func=init)
+                              interval = 25, blit = True, init_func = init)
```
@afvincent is correct.  Spaces around `=` when it is assignment, no spaces when used in a function signature / call for kwargs.

adasilva replied:
Sorry, forgot to run the pep8 checker before pushing. I've amended the commit, and it looks like it passes now.

tacaswell commented:
How did you pick up a commit from Christoph? 

@adasilva I am not sure what your familiarity with git is.  Could you rebase this onto current master and drop that first commit?

Please ask questions if you need help.  The rebase guide I like best is still a PR https://github.com/matplotlib/matplotlib/pull/2742 (I really should finish that).

If this all seems like a disaster and git fights back I can take care of it.

adasilva commented:
@tacaswell hmm, I'm not sure how that happened. I'm not very confident with rebasing, so let me check in on the process. I started following along the guide you linked to. So far, I added matplotlib/matplotlib as a remote:
`git remote add matplotlib https://github.com/matplotlib/matplotlib`
Then fetched from the remote:
`git fetch matplotlib`
Then I started to rebase:
`git rebase matplotlib/master`
This is the output:
`First, rewinding head to replay your work on top of it...
Applying: cleaned up animate_decay.py
Applying: cleaned up double_pendulum_animated.py
Applying: diversified names on vertical axis
Applying: changed while loop to for loop in animate_decay.py`
Then I looked at the status:
`On branch master
Your branch and 'origin/master' have diverged,
and have 23 and 5 different commits each, respectively.
  (use "git pull" to merge the remote branch into yours)`
Which looks a little different than the output in the doc. Am I still on the right track? In particular, it does not say `rebase in process` anywhere in my output.

adasilva commented:
Actually I might be making this more difficult. I was using rebase to edit the previous commit where there were some small mistakes, and I think that is where I picked up the previous commit. If I run
`git rebase -i HEAD~5`
then delete the line of the commit that shouldn't be there, will that do what we want?

tacaswell commented:
Yes, i think so.

On Sat, May 28, 2016, 12:37 Ashley DaSilva <notifications@github.com> wrote:

> Actually I might be making this more difficult. I was using rebase to edit
> the previous commit where there were some small mistakes, and I think that
> is where I picked up the previous commit. If I run
> git rebase -i HEAD~5
> then delete the line of the commit that shouldn't be there, will that do
> what we want?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/6486#issuecomment-222317442>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAMMhWp9mqnZJ4fjXdqsJA3OcsgDlKo5ks5qGG9kgaJpZM4IoFO_>
> .
>

discardthree commented:
@adasilva It sounds like you've got the idea, but you can grab me on facebook chat if it doesn't work. :)

adasilva commented:
I removed the extra commit.... any advice on how to resolve the new issues of the failing checks?

jenshnielsen commented:
You can ignore the test failures. Coveralls have some issues with figuring out what to compare agains (and we don't have tests of examples anyway) and the tests that failed appveyor is known to be flaky.

adasilva commented:
@jenshnielsen OK thanks!

QuLogic commented:
Replaced by #7329.
